UI/MapWnd.cpp

```diff
@@ -1939,26 +1939,93 @@ void MapWnd::RenderSystems() {
             GG::Pt circle_ul = middle - circle_half_size;
             GG::Pt circle_lr = circle_ul + circle_size;
 
+            GG::Pt circle_distance = GG::Pt(GG::X(4), GG::Y(4));
+            
+            GG::Pt outer_circle_ul = circle_ul - circle_distance;
+            GG::Pt outer_circle_lr = circle_lr + circle_distance;
+
             if (fog_scanlines
                 && (universe.GetObjectVisibilityByEmpire(system_icon.first, empire_id) <= VIS_BASIC_VISIBILITY))
             {
                 m_scanline_shader.SetColor(GetOptionsDB().Get<GG::Clr>("UI.system-fog-of-war-clr"));
                 m_scanline_shader.RenderCircle(circle_ul, circle_lr);
             }
 
-            // render circles around systems that have at least one starlane, if circles are enabled.
+            // render circles around systems that have at least one starlane if they are enabled (specific circles for player owned planets, capitals, neutrals)
             if (circles) {
                 if (std::shared_ptr<const System> system = GetSystem(system_icon.first)) {
                     if (system->NumStarlanes() > 0) {
-                        glColor(GetOptionsDB().Get<GG::Clr>("UI.unowned-starlane-colour"));
 
+                        bool has_empire_planet = false, has_neutrals = false;
+                        std::map<int, int> colony_number_by_empire_id;
+                        std::vector<std::shared_ptr<const Planet>> system_planets = Objects().FindObjects<const Planet>(system->PlanetIDs());
+                        const std::set<int>& known_destroyed_object_ids = GetUniverse().EmpireKnownDestroyedObjectIDs(HumanClientApp::GetApp()->EmpireID());
+
+                        for (std::shared_ptr<const Planet> planet : system_planets) {
+                            int planet_id = planet->ID();
+                            if (known_destroyed_object_ids.find(planet_id) != known_destroyed_object_ids.end())
```

Comment:
Use `std::set<T>::count`.  You don't need the element or the iterator so there is no need for code duplication.